index.bs

```diff
@@ -3808,18 +4263,388 @@ partial dictionary MLOpSupportLimits {
+};
+
+partial dictionary MLOpSupportLimits {
+  MLGatherSupportLimits gatherND;
+};
+</script>
+
+<div dfn-for="MLGraphBuilder/gatherND(input, indices, options)" dfn-type=argument>
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
+        - <dfn>indices</dfn>: an {{MLOperand}}. The indices array contains entire coordinates into the input tensor, with the rightmost dimension holding the number of dimensions per coordinate. So an indices tensor of shape [10,1] holds 10 single-axis indices, and a shape of [4,3] holds 4 indices of 3D coordinates. The values must be of type {{MLOperandDataType/"int32"}}, {{MLOperandDataType/"uint32"}}, or {{MLOperandDataType/"int64"}}, and each must be in the range -N (inclusive) to N (exclusive) where N is the size of the corresponding input dimension, and a negative index means indexing from the end of the corresponding dimension.
+        - <dfn>options</dfn>: an optional {{MLOperatorOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output N-D tensor of [=MLOperand/rank=] equal to the *input*'s [=MLOperand/rank=] + *indices*'s [=MLOperand/rank=] - *indices*'s [=MLOperand/shape=][-1] - 1.
+</div>
+
+<table id=constraints-gathernd class='data' link-for="MLGraphBuilder/gatherND(input, indices, options)">
+  <caption>Constraints for {{MLGraphBuilder/gatherND()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>[=/any data type|any=]</td>
+    <td>1 to [=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>{{indices}}</td>
+    <td>{{MLOperandDataType/"int32"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int64"}}</td>
+    <td>1 to [=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{input}}</td>
+    <td>*input*'s [=MLOperand/rank=] + *indices*'s [=MLOperand/rank=] - *indices*'s [=MLOperand/shape=][-1] - 1</td>
+  </tr>
+</table>
+
+{{MLOpSupportLimits}} has the following members for {{MLGraphBuilder/gatherND()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>gatherND</dfn>
+    :: Support limits for operator {{MLGraphBuilder/gatherND()}}.
+</dl>
+
+<div class="note">
+  The {{MLGraphBuilder/gatherND(input, indices, options)/indices}} parameter to {{MLGraphBuilder/gatherND()}} can not be clamped to the allowed range when the graph is built because the inputs are not known until execution. Implementations can introduce {{MLGraphBuilder/clamp()}} in the compiled graph if the required clamping behavior is not provided by the underlying platform. Similarly, if the underlying platform does not support negative indices, the implementation can introduce operations in the compiled graph to transform a negative index from the end of the dimension into a positive index.
```

Comment:
Address RFC2119 warning:
```suggestion
  The {{MLGraphBuilder/gatherND(input, indices, options)/indices}} parameter to {{MLGraphBuilder/gatherND()}} can not be clamped to the allowed range when the graph is built because the inputs are not known until execution. Implementations can introduce {{MLGraphBuilder/clamp()}} in the compiled graph if the specified clamping behavior is not provided by the underlying platform. Similarly, if the underlying platform does not support negative indices, the implementation can introduce operations in the compiled graph to transform a negative index from the end of the dimension into a positive index.
```